pkg/platform/abstract/platform.go

```diff
@@ -90,6 +93,15 @@ func (ap *Platform) HandleDeployFunction(createFunctionOptions *platform.CreateF
 		return nil, errors.Wrap(err, "Failed determining whether function should build")
 	}
 
+	// special case when we are asked to build the function and it wasn't been created yet
```

Comment:
doesn't this logic clash with the clause that will pass existingFunctionConfig nil if metadata isn't passed? user might pass only spec with never build and you'll fail it, even though the function exists